File: Queue_reconstruct.java
// Time Complexity :
// Space Complexity :
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        Arrays.sort(people, ( a, b) -> {
            if(a[0] == b[0]){
                return a[1]-b[1];
            }
            return b[0]- a[0];
        });
        ArrayList<int[]> output = new ArrayList<>();
        
        for(int[] p: people)
            output.add(p[1],p);
        return output.toArray(new int[0][]);
       
    }
}

File: QueueReconstruct.py
#406. Queue Reconstruction by Height
"""
Time Complexity : O(n^2)
Space Complexity : O(n) #overwriting people array for sorting.
"""
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        #sort according to h(i)
        people = sorted(people, key = lambda x: (-x[0], x[1])) #n log(n)
        print(people)
        
        result = list()
        
        #re-arrange according to k(i)
        for i in range(0, len(people)):
            hi, ki = people[i]
            print(hi, ki)
            result.insert(ki, people[i])
            
        return result
